arteria/models/runfolder_utils.py

```diff
@@ -0,0 +1,177 @@
+        self._state_file = (self.path / ".arteria/state")
+        if not self._state_file.exists():
+            self._state_file.write_text("ready")
+
+    @property
+    def state(self):
+        return State(self._state_file.read_text().strip().lower())
+
+    @state.setter
+    def state(self, new_state):
+        assert new_state in State
+        self._state_file.write_text(new_state.value)
+
+    @property
+    def metadata(self):
+        if not self.run_parameters:
+            log.warning(f"No metadata found for runfolder {self.path}")
+
+        metadata = {}
+
+        try:
+            metadata["reagent_kit_barcode"] = \
+                self.run_parameters["ReagentKitBarcode"]
+        except KeyError:
+            log.debug("Reagent kit barcode not found")
+
+        try:
+            metadata["library_tube_barcode"] = \
+                self.run_parameters["RfidsInfo"]["LibraryTubeSerialBarcode"]
+        except KeyError:
+            try:
+                metadata["library_tube_barcode"] = \
+                    next(
+                        consumable["SerialNumber"]
+                        for consumable in self.run_parameters["ConsumableInfo"]["ConsumableInfo"]
+                        if consumable["Type"] == "SampleTube"
+                    )
+            except (KeyError, StopIteration):
+                log.debug("Library tube barcode not found")
+
+        return metadata
+
+
+class Instrument:
+    RUNPARAMETERS_INSTRUMENT_ID_KEYS = [
+        "InstrumentName", "InstrumentId", "ScannerID",
+        "InstrumentSerialNumber"
+    ]
+
+    INSTRUMENT_MARKER_DICT = {
+        """
+        Returns a dict of instruments with the following details
+            id_pattern: id pattern that identifies the Instrument
+            completed_marker_file: File name(str) of the completed marker file
+
+        """
+        "NovaSeq": {
+            'id_pattern': '^A', 'completed_marker_file': 'CopyComplete.txt'
+        },
+        "NovaSeqXPlus": {
+            'id_pattern': '^LH', 'completed_marker_file': 'CopyComplete.txt'
+        },
+        "ISeq": {
+            'id_pattern': '^FS', 'completed_marker_file': 'CopyComplete.txt'
+        },
+        "MiSeq": {
+            'id_pattern': '^M', 'completed_marker_file': 'RTAComplete.txt'
+        },
+        "HiSeq": {
+            'id_pattern': '^D', 'completed_marker_file': 'RTAComplete.txt'
+        },
+        "HiSeqX": {
+            'id_pattern': '^ST-E', 'completed_marker_file': 'RTAComplete.txt'
+        }
+    }
+
+    def __init__(self, run_parameters):
+        self.run_parameters = run_parameters
+        assert self.run_parameters
+
+    @property
+    def completed_marker_file(self):
+        """
+        Returns the completed_marker_file name(str) which is specific to the instrument
+        """
+        instrument, instrument_keys = self.instrument
+        completed_marker_file = instrument_keys.get("completed_marker_file")
```

Comment:
```suggestion
        completed_marker_file = instrument_keys["completed_marker_file"]
```
This function should always return a valid marker file (and never `None`).